packages/devtools_server/lib/src/server.dart

```diff
@@ -233,3 +269,20 @@ void printOutput(
 }) {
   print(machineMode ? jsonEncode(json) : message);
 }
+
+/// Map the URI (which may already be Observatory web app) to a WebSocket URI
+/// for the VM service.
+///
+/// If the URI is already a VM Service WebSocket URI it will not be modified.
+Uri getVmServiceUriFromObservatoryUri(Uri uri) {
```

Comment:
[nit] https://www.dartlang.org/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

Reply:
I should re-read that doc periodically till it sinks in =)

(I'm surprised we don't have a lint for this one - at least, I can't see one)

Reply:
@pq We should have a lint and quick fix that catches zero argument methods where the name starts with "get" and quick fixes them to be actual getters.